publication/models/distribution_list.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014-2018 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+SQL_CONTRACT_COUNT = """
+SELECT COALESCE(SUM(ROUND(l.quantity)::integer), 0) as quantity
+ FROM account_analytic_invoice_line l
+ JOIN account_analytic_account c
+ ON l.analytic_account_id = c.id
+ WHERE l.product_id = %s
+ AND c.partner_id = %s"""
+
+SQL_ASSIGNED_COUNT = """
+SELECT COALESCE(SUM(copies), 0) as copies
+ FROM distribution_list
+ WHERE product_id = %s
+ AND contract_partner_id = %s"""
+
+
+class DistributionList(models.Model):
```

Comment:
Why not `PublicationList`?
It is a bit confusing to use Distribution and Publication everywhere

Reply:
It is DistributionList because it is not a list of publications, but rather a list about how publications should be distributed over recipients.

Reply:
I'd still propose to rename this to `publication.distribution.list` to avoid possible clashes with other model names that are distribution lists, as that's a quite generic concept. Personally, I usually prefix all model names an addon introduces with the addon's name, as those have to be unique anyways. Produces sometimes horribly long model names, but it unique.